ps2/hangman.py

```diff
@@ -104,85 +101,90 @@ def get_available_letters(letters_guessed):
+    It decides if you won and your score or it tells you you lost.
     '''
-    global letters_guessed
     letters_guessed = []
     guesses_remaining = 6
+    warnings_remaining = 3
     print('Welcome to the game Hangman!')
     print('I am thinking of a word that is {} letters '
           'long.'.format(len(secret_word)))
-    print('---------------------------------------')
 
     while guesses_remaining > 0:
+        print('---------------------------------------')
         print('You have {} guesses left'.format(guesses_remaining))
         print('Available letters: {}'.format(get_available_letters(
                                              letters_guessed)))
-        good_guess = letter_input()
-        if good_guess:
+# Getting Input
+        guess_a_letter = input('Please guess a letter: ')
+        guess_a_letter = str.lower(guess_a_letter)
+# Checking if Input is Valid
+        if not str.isalpha(guess_a_letter):
+            if warnings_remaining == 0:
+                guesses_remaining -= 1
+                print('''Oops! That is not a valid letter and you have no warnings
+left so you lose one guess: ''',
+                      get_guessed_word(secret_word, letters_guessed))
+                if guesses_remaining <= 0:
+                    print('Sorry, you ran out of guesses. The word was', secret_word)
+                continue
+            if warnings_remaining > 0:
+                warnings_remaining -= 1
+                print('''Oops! That is not a valid letter. You have {} warnings left:
+'''.format(warnings_remaining), get_guessed_word(secret_word, letters_guessed))
+                continue
+        if guess_a_letter in letters_guessed:
+            if warnings_remaining == 0:
+                guesses_remaining -= 1
+                print('''Oops! You've already guessed that letter and you have
+0 warnings left so you lose one guess: ''',
+                      get_guessed_word(secret_word, letters_guessed))
+                if guesses_remaining <= 0:
+                    print('Sorry, you ran out of guesses. The word was', secret_word)
+                continue
+            if warnings_remaining > 0:
+                warnings_remaining -= 1
+                print('''Oops! You've already guessed that letter.
+    You now have {} warnings: '''.format(warnings_remaining),
+                    get_guessed_word(secret_word, letters_guessed))
+                continue
+# Checking if the Guess Was Good or Bad
+        if guess_a_letter in secret_word:
             letters_guessed.append(guess_a_letter)
+            print('Good Guess: ', get_guessed_word(
+                                  secret_word, letters_guessed))
+
+        if guess_a_letter not in secret_word:
+            print('Oops! That letter is not in my word: ', get_guessed_word(
+                                  secret_word, letters_guessed))
+            if (guess_a_letter == 'a' or guess_a_letter == 'e' or guess_a_letter == 'i'
```

Comment:
```python
if guess_a_letter in 'aeiou':
```

Are guesses guaranteed to be lower case?

Reply:
yes, cuz of line 126.